index.bs

```diff
@@ -5069,96 +5069,125 @@ No [=extended attributes=] are applicable to dictionaries.
+[=exception/message=].
 
-    Here is are some examples of wording to use to create and throw exceptions.
-    To throw a new [=simple exception=] named {{TypeError}}:
+<div class="example" id="example-exception-creation-and-throwing">
+    <p>Here is are some examples of wording to use to create and throw exceptions.
+    To throw a new [=simple exception=] whose type is {{TypeError}}:
 
     <blockquote>
-        [=exception/Throw=] a {{TypeError}}.
+        <p>[=exception/Throw=] a {{TypeError}}.
     </blockquote>
 
-    To throw a new {{DOMException}} with [=error name=] "{{NotAllowedError!!exception}}":
+    <p>To throw a new {{DOMException}} with [=error name=] "{{NotAllowedError}}":
 
     <blockquote>
-        [=exception/Throw=] a "{{NotAllowedError!!exception}}" {{DOMException}}.
+        <p>[=exception/Throw=] an "{{NotAllowedError}}" {{DOMException}}.
     </blockquote>
 
-    To create a new {{DOMException}} with [=error name=] "{{SyntaxError!!exception}}":
+    <p>To create a new {{DOMException}} with [=error name=] "{{SyntaxError}}":
 
     <blockquote>
-        Let <var ignore>object</var> be a newly [=created=] "{{SyntaxError!!exception}}" {{DOMException}}.
+        <p>Let <var ignore>object</var> be a newly [=exception/created=] "{{SyntaxError}}"
+        {{DOMException}}.
+    </blockquote>
+
+    <p>To [=reject=] a promise with a new {{DOMException}} with [=error name=] "{{OperationError}}":
+
+    <blockquote>
+        [=Reject=] <var ignore>p</var> with an "{{OperationError}}" {{DOMException}}.
     </blockquote>
 </div>
 
+<div class=example id=example-exception-throwing-message-hint>
+    <p>An example of including additional information used to construct the [=exception/message=]
+    would be:
 
-<h4 id="idl-DOMException-error-names">Error names</h4>
+    <blockquote>
+        [=exception/Throw=] a "{{SyntaxError}}" {{DOMException}} indicating that the given value
+        had disallowed trailing spaces.
+    </blockquote>
 
-The <dfn id="dfn-error-names-table" export>error names table</dfn> below lists all the allowed error names
-for {{DOMException}}, a description, and legacy code values.
+    <p>Such additional context is most helpful to implementers when it is not immediately obvious
+    why the exception is being thrown, e.g., because there are many different steps in the algorithm
+    which throw a "{{SyntaxError}}" {{DOMException}}. In contrast, if your specification throws a
+    "{{NotAllowedError}}" {{DOMException}} immediately after checking if the user has provided
+    permission to use a given feature, it's fairly obvious what sort of [=exception/message=] the
+    implementation should construct, and so specifying it is not necessary.
+</div>
 
-<p class="warning">
-    The {{DOMException}} names marked as deprecated are kept for legacy purposes but their usage is discouraged.
-</p>
+The resulting behavior from creating and throwing an exception is language binding-specific.
```

Comment:
```suggestion
The resulting behavior from creating and throwing an exception is language-binding specific.
```
(Or maybe hyphens everywhere?)

Reply:
It looks like we have many instances of no hyphens at all, so I'll switch to that. I get the sense this is one of those questions that vary by style guide, but I can't seem to find the right combination of Google keywords to pin down any specific style guides for what's happening here (which is a noun phrase being used to modify an adjective).